flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV120.java

```diff
@@ -168,4 +175,44 @@ public CatalogColumnStatisticsDataDate toFlinkDateColStats(ColumnStatisticsData
 			throw new CatalogException("Failed to create Flink statistics for date column", e);
 		}
 	}
+
+	@Override
+	public Set<String> listBuiltInFunctions() {
+		try {
+			Method method = FunctionRegistry.class.getMethod("getFunctionNames");
+			// getFunctionNames is a static method
+			Set<String> names = (Set<String>) method.invoke(null);
+
+			return names.stream()
+				.filter(n -> isBuiltInFunctionInfo(getFunctionInfo(n)))
+				.collect(Collectors.toSet());
+		} catch (Exception ex) {
+			throw new CatalogException("Failed to invoke FunctionRegistry.getFunctionNames()", ex);
+		}
+	}
+
+	@Override
+	public Optional<FunctionInfo> getBuiltInFunctionInfo(String name) {
+		FunctionInfo functionInfo = getFunctionInfo(name);
+
+		return isBuiltInFunctionInfo(functionInfo) ? Optional.of(functionInfo) : Optional.empty();
```

Comment:
What happens if `functionInfo` is null, i.e. the function under the name doesn't exist?

Reply:
good catch. I fixed that part, and added UT. Pls take another look